src/parse-stream.js

```diff
@@ -567,6 +567,51 @@ export default class ParseStream extends Stream {
         this.trigger('data', event);
         return;
       }
+      match = (/^#EXT-X-DATERANGE:(.*)$/).exec(newLine);
+      if (match && match[1]) {
+        event = {
+          type: 'tag',
+          tagType: 'daterange'
+        };
+        event.attributes = parseAttributes(match[1]);
+        ['ID', 'CLASS'].forEach(function(key) {
+          if (event.attributes.hasOwnProperty(key)) {
+            event.attributes[key] = String(event.attributes[key]);
+          }
+        });
+        ['START-DATE', 'END-DATE'].forEach(function(key) {
+          if (event.attributes.hasOwnProperty(key)) {
+            event.attributes[key] = new Date(event.attributes[key]);
+          }
+        });
+        ['DURATION', 'PLANNED-DURATION'].forEach(function(key) {
+          if (event.attributes.hasOwnProperty(key)) {
+            event.attributes[key] = parseFloat(event.attributes[key]);
+          }
+        });
+
+        ['SCTE35-CMD', ' SCTE35-OUT', 'SCTE35-IN'].forEach(function(key) {
+          if (event.attributes.hasOwnProperty(key)) {
+            event.attributes[key] = parseInt(event.attributes[key], 16);
+          }
+        });
+
+        const clientAttributePattern = /^X-([A-Z]+-)+[A-Z]+$/;
```

Comment:
Is it worth adding some tests for these client attributes?

Reply:
Done!